crate/azure_cvm/src/imds.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::str::FromStr;
+
+use crate::{attestation_report::TdReport, error::Error};
+
+use base64::{engine::general_purpose, Engine as _};
+use zerocopy::AsBytes;
+
+// IMDS endpoint for VCEK certificate, AMD SEV CA and AMD Root CA
+const IMDS_THIM_ENDPOINT: &str = "http://169.254.169.254/metadata/THIM/amd/certification";
+// IMDS endpoint for Intel TD quote
+const IMDS_TDX_ENDPOINT: &str = "http://169.254.169.254/acc/tdquote";
+
+/// Retrieves a TDX quote from the Azure Instance Metadata Service (IMDS) using a provided TD
+/// report.
+pub fn get_td_quote(td_report: &TdReport) -> Result<Vec<u8>, Error> {
+    let payload = serde_json::json!({"report": general_purpose::URL_SAFE_NO_PAD.encode(td_report.as_bytes())});
+    let url =
+        reqwest::Url::from_str(IMDS_TDX_ENDPOINT).map_err(|e| Error::BadURLError(e.to_string()))?;
+
+    let r = reqwest::blocking::Client::builder()
```

Comment:
```suggestion
    let response = reqwest::blocking::Client::builder()
```
not totally convinced, discard it if you want

Reply:
Fixed in 4767391487ed7165d356b439f270d4b3cf1d3cdc.